### PR TITLE
Fixed #27655 -- Added some guidelines to the coding style docs.

### PR DESCRIPTION
**Pull request by timgraham**
I'm going to leave this open for a little while and add other things that I came across as I do reviews.
https://code.djangoproject.com/ticket/27655

### CODE REVIEW

**Comment by sir-sigurd** [comment]
I think this could need some clarification too:
```python
with self.assertRaisesMessage(
    SomeError,
    'very '
    'long '
    'message.'
):
```
vs.
```python
msg = (
    'very '
    'long '
    'message.'
)
with self.assertRaisesMessage(SomeError, msg):
```

File: docs/internals/contributing/writing-code/coding-style.txt [hunk 1]
```diff
@@ -33,6 +33,27 @@ Python style
...
+
+      raise AttributeError("Here is a multine error message "
+                           "shortened for clarity.")
+
+  This makes better use of space and avoids having to realign strings if the
+  length of the first line changes.
+
+* Use single quotes for strings, or a double quote if the the string contains a
+  single quote. Don't waste time doing unrelated refactoring of existing code
+  to confirm to this style.
```
Comment (jschneier) [inline docs/internals/contributing/writing-code/coding-style.txt]: Typo, should be `conform`

File: docs/internals/contributing/writing-code/coding-style.txt [hunk 2]
```diff
@@ -33,6 +33,27 @@ Python style
 
 * Use four spaces for indentation.
 
+* Use four space hanging indentation rather than vertical alignment::
+
+    raise AttributeError(
+        "Here is a multine error message "
+        "shortened for clarity."
```
Comment (adamchainz) [inline docs/internals/contributing/writing-code/coding-style.txt]: this should be single quotes as per the next bullet point 😉

**Comment by adamchainz** [comment]
This has been open some time now, merge it and create another if any more rules crop up?

**Pull request merged**